From a8c4375a01c55eaef53e457a045e9874fc930d88 Mon Sep 17 00:00:00 2001
From: Matt Clarkson <mattyclarkson@gmail.com>
Date: Fri, 15 Mar 2024 12:10:45 +0000
Subject: [PATCH] Add pure tests for mathmatical libraries

The libraries have `noasm` support. These tests validate that support does not regress.
---
 dh/x25519/BUILD.bazel | 16 ++++++++++++++++
 dh/x448/BUILD.bazel   | 16 ++++++++++++++++
 2 files changed, 32 insertions(+)

diff --git a/dh/x25519/BUILD.bazel b/dh/x25519/BUILD.bazel
index cacfcf1..7464c66 100644
--- a/dh/x25519/BUILD.bazel
+++ b/dh/x25519/BUILD.bazel
@@ -41,3 +41,19 @@ go_test(
         "//math/fp25519",
     ],
 )
+
+go_test(
+    name = "x25519_pure_test",
+    srcs = [
+        "curve_test.go",
+        "key_test.go",
+    ],
+    data = glob(["testdata/**"]),
+    embed = [":x25519"],
+    pure = "on",
+    deps = [
+        "//internal/conv",
+        "//internal/test",
+        "//math/fp25519",
+    ],
+)
diff --git a/dh/x448/BUILD.bazel b/dh/x448/BUILD.bazel
index 94a7b2b..b96c160 100644
--- a/dh/x448/BUILD.bazel
+++ b/dh/x448/BUILD.bazel
@@ -41,3 +41,19 @@ go_test(
         "//math/fp448",
     ],
 )
+
+go_test(
+    name = "x448_pure_test",
+    srcs = [
+        "curve_test.go",
+        "key_test.go",
+    ],
+    data = glob(["testdata/**"]),
+    embed = [":x448"],
+    pure = "on",
+    deps = [
+        "//internal/conv",
+        "//internal/test",
+        "//math/fp448",
+    ],
+)
